src/axom/sidre/core/Group.hpp

```diff
@@ -272,57 +283,90 @@ class Group
+  umpire::Allocator getDefaultArrayAllocator() const
   {
     umpire::ResourceManager& rm = umpire::ResourceManager::getInstance();
-    return rm.getAllocator(m_default_allocator_id);
+    return rm.getAllocator(m_default_array_alloc_id);
   }
 
-  /*!
-   * \brief Set the default umpire::Allocator associated with this Group.
-   */
-  Group* setDefaultAllocator(umpire::Allocator alloc)
+  //! \brief Set the default array umpire::Allocator associated with this Group.
+  Group* setDefaultArrayAllocator(umpire::Allocator alloc)
   {
-    m_default_allocator_id = alloc.getId();
+    setDefaultAllocator(alloc.getId());
     return this;
   }
+#endif
 
   /*!
-   * \brief Set the default umpire::Allocator associated with this Group.
-   */
-  Group* setDefaultAllocator(int allocId)
+   * \brief Set the default array allocator id associated with this Group.
+   *
+   * This allocator is the default for array data, even if the array is
+   * length 1.  (Note, tuples are not arrays in this specific sense.  @see
+   * setDefaultTupleAllocator(int).)
+  */
+  Group* setDefaultArrayAllocator(int allocId)
   {
-    m_default_allocator_id = allocId;
+    SLIC_ASSERT(allocId != axom::INVALID_ALLOCATOR_ID);
+    m_default_array_alloc_id = allocId;
     return this;
   }
-#else
-  /*!
-   * \brief Return the ID of the default umpire::Allocator associated with this
-   * Group.
-   */
-  int getDefaultAllocatorID() const { return axom::getDefaultAllocatorID(); }
+
+  //! \brief Return the ID of the default scalar/tuple allocator id associated with this Group.
+  int getDefaultTupleAllocatorID() const { return m_default_tuple_alloc_id; }
+
+#if defined(AXOM_USE_UMPIRE)
+  //! \brief Return the default scalar/tuple umpire::Allocator associated with this Group.
+  umpire::Allocator getDefaultTupleAllocator() const
+  {
+    umpire::ResourceManager& rm = umpire::ResourceManager::getInstance();
+    return rm.getAllocator(m_default_tuple_alloc_id);
+  }
+
+  //! \brief Set the default scalar/tuple umpire::Allocator associated with this Group.
+  Group* setDefaultTupleAllocator(umpire::Allocator alloc)
+  {
+    setDefaultTupleAllocator(alloc.getId());
+    return this;
+  }
+#endif
 
   /*!
-   * \brief Set the default umpire::Allocator associated with this Group.
-   */
-  Group* setDefaultAllocator(int allocId)
+   * \brief Set the default scalar/tuple allocator id associated with this Group.
+   *
+   * This allocator is the default for tuple data, including strings
+   * and scalars.  (Arrays are not tuples in this sense.  @see
+   * setDefaultArrayAllocator(int).)
+  */
+  Group* setDefaultTupleAllocator(int allocId)
   {
-    AXOM_UNUSED_VAR(allocId);
-    SLIC_ASSERT(allocId == axom::getDefaultAllocatorID());
+    SLIC_ASSERT(allocId != axom::INVALID_ALLOCATOR_ID);
+    m_default_tuple_alloc_id = allocId;
+    ConduitMemory::instanceForAxomId(m_default_tuple_alloc_id).conduitId();
```

Comment:
Is this a global conduit call within a local group in sidre?
(or am I misunderstanding it)

Reply:
It's that.